test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java

```diff
@@ -222,7 +251,111 @@ protected final XContentBuilder fieldMapping(CheckedConsumer<XContentBuilder, IO
         });
     }
 
-    QueryShardContext createQueryShardContext(MapperService mapperService) {
+    private AggregationContext aggregationContext(MapperService mapperService, IndexSearcher searcher, Query query) {
```

Comment:
I decided not to go with mockito here partially because I wanted to suffer every time I added a new method to AggreationContext.

Reply:
I can't tell if you're joking or not.

Reply:
I'm really not. Suffering makes you think "should I really add this method? this class is already big. maybe there is a cleaner way."